### PR TITLE
[Bugfix] Fix refresh not failed after modify table column type

### LINKED ISSUE #9895: [sync materialized view]refresh not failed after modify table column type
### Steps to reproduce the behavior (Required)
```
CREATE TABLE `t2` (
                                            `k1` date,
                                            `k2` int,
                                            `k3` int
                                        ) ENGINE=OLAP
                                        DUPLICATE KEY(`k1`)
                                        COMMENT "OLAP"
                                        PARTITION BY RANGE (k1) (
                                        START ("2020-10-01") END ("2022-03-04") EVERY (INTERVAL 15 day)
                                        )
                                        DISTRIBUTED BY HASH(`k1`) BUCKETS 3;
CREATE MATERIALIZED VIEW mv2
                                     PARTITION BY k1
                                     DISTRIBUTED BY HASH(k1) BUCKETS 10
                                     REFRESH ASYNC
                                     AS SELECT k1, k2
                                     FROM t2;
ALTER TABLE t2 MODIFY COLUMN k2 VARCHAR(20);
INSERT INTO t2 VALUES ("2020-10-10",'test_str',1);
SELECT * FROM INFORMATION_SCHEMA.task_runs order by create_time desc limit 1;
```

### Expected behavior (Required)
`task failed`
### Real behavior (Required)
`task succeed`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.4-RELEASE f823e8b |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by QingdongZeng3**
[Bugfix] fix refresh not failed after modify table column type

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9895

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 23 (26.09%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|6|23|26.09%|[573, 574, 575, 576, 577, 579, 580, 581, 588, 589, 590, 591, 592, 593, 595, 596, 597]|

**Comment by QingdongZeng3** [comment]
@Astralidea PTAL

**Comment by QingdongZeng3** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java [hunk 1]
```diff
@@ -560,6 +566,36 @@ protected void runRunningJob() throws AlterCancelException {
...
+
+    private void inactiveRelatedMv(Set<String> modifiedColumns, OlapTable tbl) {
+        if (modifiedColumns.isEmpty()) {
+            return;
+        }
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+        for (Long mvId : tbl.getRelatedMaterializedViews()) {
+            for (Column mvColumn : db.getTable(mvId).getColumns()) {
+                if (modifiedColumns.contains(mvColumn.getName())) {
+                    MaterializedView mv = (MaterializedView) db.getTable(mvId);
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java:592]: should check mv is null

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 26 (23.08%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|6|26|23.08%|[573, 574, 575, 576, 577, 579, 580, 581, 588, 589, 590, 591, 592, 593, 595, 596, 597, 599, 600, 601]|

**Review by Astralidea — DISMISSED** [review]

**Review by ABingHuang — CHANGES_REQUESTED** [review]
could you add an UT for it?

**Comment by QingdongZeng3** [comment]
run starrocks_admit_test

**Comment by QingdongZeng3** [comment]
> could you add an UT for it?

ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 24 / 26 (92.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|24|26|92.31%|[592, 593]|

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**